pkg/bgpv1/manager/reconcile.go

```diff
@@ -649,49 +643,85 @@ func hasLocalEndpoints(svc *slim_corev1.Service, ls localServices) bool {
 // fullReconciliation reconciles all services, this is a heavy operation due to the potential amount of services and
 // thus should be avoided if partial reconciliation is an option.
 func (r *LBServiceReconciler) fullReconciliation(ctx context.Context, sc *ServerWithConfig, newc *v2alpha1api.CiliumBGPVirtualRouter, ls localServices) error {
-	// Loop over all existing announcements, delete announcements for services which no longer exist
+	toReconcile, toWithdraw, err := r.fullReconciliationServiceList(sc)
+	if err != nil {
+		return err
+	}
+	for _, svc := range toReconcile {
+		if err := r.reconcileService(ctx, sc, newc, svc, ls); err != nil {
+			return fmt.Errorf("failed to reconcile service %s/%s: %w", svc.Namespace, svc.Name, err)
+		}
+	}
+	for _, svc := range toWithdraw {
+		if err := r.withdrawService(ctx, sc, svc); err != nil {
+			return fmt.Errorf("failed to withdraw service %s/%s: %w", svc.Namespace, svc.Name, err)
+		}
+	}
+	return nil
+}
+
+// svcDiffReconciliation performs reconciliation, only on services which have been created, updated or deleted since
+// the last diff reconciliation.
+func (r *LBServiceReconciler) svcDiffReconciliation(ctx context.Context, sc *ServerWithConfig, newc *v2alpha1api.CiliumBGPVirtualRouter, ls localServices) error {
+	toReconcile, toWithdraw, err := r.diffReconciliationServiceList()
+	if err != nil {
+		return err
+	}
+	for _, svc := range toReconcile {
+		if err := r.reconcileService(ctx, sc, newc, svc, ls); err != nil {
+			return fmt.Errorf("failed to reconcile service %s/%s: %w", svc.Namespace, svc.Name, err)
+		}
+	}
+	// Loop over the deleted services
+	for _, svcKey := range toWithdraw {
+		if err := r.withdrawService(ctx, sc, svcKey); err != nil {
+			return fmt.Errorf("failed to withdraw service %s: %w", svcKey, err)
+		}
+	}
+	return nil
+}
+
+// fullReconciliationServiceList return a list of services to reconcile and to withdraw when performing
+// full service reconciliation.
+func (r *LBServiceReconciler) fullReconciliationServiceList(sc *ServerWithConfig) (toReconcile []*slim_corev1.Service, toWithdraw []resource.Key, err error) {
```

Comment:
Do some of these methods need to be exportable ?

Reply:
it is not required, resolving it.

Reply:
nope, not for now